pkg/daemon/ceph/osd/key_rotation.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package osd
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rook/rook/cmd/rook/rook"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/osd/kms"
+	oposd "github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+)
+
+const (
+	slotZero string = "0"
+	slotOne  string = "1"
+)
+
+func RotateEncryptionKey(kms *kms.Config, context *clusterd.Context, secretName string, devicePaths []string) error {
+	logger.Info("fetching the current key")
+	// Fetch the currentKey.
+	currentKey, err := kms.GetSecret(secretName)
+	if err != nil {
+		rook.TerminateFatal(errors.Wrapf(err, "failed to get secret %q", secretName))
```

Comment:
question: is this right to use `TerminateFatal`? since it will do exit 1.

Reply:
missed it while moving it to osd package, 
changed to return